Android Source/mobile/src/main/java/de/jo2k/loggingmodule/DataHandler.java

```diff
@@ -0,0 +1,138 @@
+
+import static net.steppschuh.sensordatalogger.PhoneApp.TAG;
+
+public class DataHandler {
+    private boolean isRecording = false;
+    private FileWriter filewriter;
+    private String filename;
+    private boolean firstEntryWasWritten;
+
+    public void logDataBatch(DataBatch dataBatch, String sourceNodeId) {
+        /**
+         * Writes newly obtained data to the logfile.
+         * User should call DataHandler.startRecording() beforehand to initialize a file.
+         * Returns silently if no recording was started beforehand.
+         */
+        if (!isRecording)
+            return;
+
+        // Log.v(TAG, sourceNodeId);
+        // Log.v(TAG, dataBatch.toString());
+
+        try {
+            // write the comma only if we added some entries before
+            if (firstEntryWasWritten) {
+                filewriter.write(",");
+            }
+            firstEntryWasWritten = true;
+            filewriter.write(dataBatch.toJson());
+        } catch (IOException e) {
+            Log.e(TAG, "FileWriter could not write.");
+            e.printStackTrace();
+        }
+    }
+
+    public boolean isRecording() {
+        /**
+         * Returns the status of the DataHandler, true if a recording is in progress, false otherwise.
+         */
+        return isRecording;
+    }
+
+    public void startRecording(String filename) {
+        /**
+         * Starts the recording of sensors to a file denoted by `filename` by initializing a new
+         * file on the external storage.
+         */
+
+        if (isRecording) {
+            stopRecording();
+        }
+
+        // set recording status
+        isRecording = true;
+        firstEntryWasWritten = false;
+
+        // set path
+        String path = Environment.getExternalStorageDirectory() + "/" + BuildConfig.APPLICATION_ID + "/";
+
+        // create folder
+        File folder = new File(path);
+        if (!folder.exists()) {
+            boolean success = folder.mkdirs();
+            if (success) {
+                Log.v(TAG, "DataHandler created folder " + folder);
+            } else {
+                Log.e(TAG, "DataHandler could not create folder " + folder);
+                isRecording = false;
+                return;
+            }
+        }
+
+        // determine final path to write to
+        this.filename = path + filename + ".json";
+        Log.v(TAG, "DataHandler records to file: " + this.filename);
+
+        // Create file and filewriter
+        File file = new File(this.filename);
+        FileWriter filewriter = null;
+
+        // initialize filewriter
+        try {
+            filewriter = new FileWriter(file, true);
+            filewriter.write("[");
+        } catch (IOException e) {
+            Log.e(TAG, "Could not create filewriter in DataHandler.startRecording()");
+            e.printStackTrace();
+        }
+
+        // assign local variables
+        this.filewriter = filewriter;
+    }
+
+    public String stopRecording () {
```

Comment:
Similar to `startRecording`, I would return the created `File` here. Or make it `void` and add a getter for the most recently created file.

Reply:
:+1: (Included both suggestions)